### ISSUE TITLE
Regression issue in the wasmer3 branch when using qjs.wasm

### ISSUE THREAD

**Issue by john-sharratt** [wasmerio/wasmer#2865]
### Describe the bug

Spent about 6 hours trying to trace down a regression bug thinking it was my merge but it seems this commit did it...
[e4afb8f821222d8b7ea686889e1e8822a46d66f3](https://github.com/wasmerio/wasmer/commit/e4afb8f821222d8b7ea686889e1e8822a46d66f3)

No idea whats causing it and tried a ton of things but I just can't find the cause.

Steps to reproduce...

```
john@LappyWorld:/prog/wasmer/lib/cli$ cargo run --features debug,singlepass -- /prog/ate/tokweb/public/bin/qjs.wasm
    Finished dev [unoptimized + debuginfo] target(s) in 49.06s
     Running `/prog/wasmer/target/debug/wasmer /prog/ate/tokweb/public/bin/qjs.wasm`
QuickJS - Type "\h" for help
qjs > 1 + 2
^C
```

It's meant to do this....

```
john@LappyWorld:/prog/wasmer/lib/cli$ cargo run --features debug,singlepass -- /prog/ate/tokweb/public/bin/qjs.wasm
    Finished dev [unoptimized + debuginfo] target(s) in 53.74s
     Running `/prog/wasmer/target/debug/wasmer /prog/ate/tokweb/public/bin/qjs.wasm`
QuickJS - Type "\h" for help
qjs > 1 + 2
1 + 2
3
qjs > ^C
```

From what I can tell qjs polling a clock but never leaves that loop.
You can get qjs.wapm from WAPM.

**Comment by epilys** [comment]
It seems that in `lib/wasi/src/syscalls/legacy/snapshot0.rs`, function `poll_oneoff`, a subscription event of type read (`type_ == 1 == __WASI_EVENTTYPE_FD_READ`), for fd ` == 0 == stdin` is passed into `lib/wasi/src/syscalls/mod.rs`, function `poll_oneoff` but then the event has (`type_ == 0 == __WASI_EVENTTYPE_CLOCK`).

**Comment by Amanieu** [comment]
This was fixed by #2870.
